azureDatabaseUtils/src/main/java/bio/terra/workspace/azureDatabaseUtils/database/DatabaseService.java

```diff
@@ -80,4 +90,157 @@ public void restoreNamespaceRoleAccess(String namespaceRole) {
 
     databaseDao.restoreLoginPrivileges(namespaceRole);
   }
+
+  public void pgDump(
+      String dbName,
+      String dbHost,
+      String dbPort,
+      String adminUser,
+      String blobFileName,
+      String blobContainerName,
+      String blobContainerUrlAuthenticated) {
+
+    // Grant the database role (dbName) to the landing zone identity (adminUser).
+    // In theory, we should be revoking this role after the operation is complete.
+    // We are choosing to *not* revoke this role for now, because:
+    // (1) we could run into concurrency issues if multiple users attempt to clone the same
+    // workspace at once;
```

Comment:
users should be able to clone at once thought, so I am not sure I understand how identity impacts that but maybe I missed another piece here?

Reply:
The problem is when there are concurrent clone operations. If one operation removes access, it may cause the other process to fail. Cloning operations are independent so understanding when to remove access is difficult.

Reply:
ah like you mean if someone changes permissions to a given workspace or something like that? would it make sense to give the cloning operation the necessary permission for the whole e2e it needs to complete as long as it had them at cloning request?

Reply:
I'm not sure how to actually implement that idea in a way that would help avoid the case where some independent operation removes access at an inopportune moment...

Reply:
Im sorry but I still dont quite understand why 2 users cloning at the same time is not possible. With smaller data sets this wont be an issue since cloning happens fast but with larger ones (say where the db dump is more then say 1GB, possibly even smaller) not being to have two users clone at once may cause issues that may cause cloning to fail or cause the user to wait a long time for their workspace to be ready

Reply:
No worries, this thread has become a bit confusing: two users cloning simultaneously *would* be a problem if we revoke the role after the operation is complete, because there's a chance that one user's cloning process would remove the role's permission while another one is trying to operate on the database. 

That's why we're choosing not to revoke the role.

Reply:
aha, okay. I think I finally follow :) thanks for the clarification! I missed the part where this only realted to revoking the role somehow.